test/generate_test_sam.cxx

```diff
@@ -87,6 +76,27 @@ int main(int argc, char* argv[]) {
     }
     
     out.close();
+}
+
+static void BM_GenerateSAM(benchmark::State& state) {
+    for (auto _ : state) {
+        GenerateSAMFile("benchmark_temp.sam", state.range(0));
+        std::remove("benchmark_temp.sam");
+    }
+}
+
+BENCHMARK(BM_GenerateSAM)->Arg(100);
```

Comment:
We need check how this scales. Maybe we should take an external benchmark parameter up to how many entries and then we have a benchmark range that checks this...